internal/controller/clusterinstance_controller.go

```diff
@@ -666,21 +706,134 @@ func (r *ClusterInstanceReconciler) applyRenderedManifests(
 	return
 }
 
+// getResourcesToPrune builds an ordered list of resources to prune from the rendered object map
+//
+//	resources are ordered in the following way:
+//	1. descending order of sync-wave
+//	2. for resources with the same sync-wave, they are alphabetically sorted
+func getResourcesToPrune(groupedRenderedObjs MapRenderedObjects) ([]ci.RenderedObject, error) {
```

Comment:
Instead of getting this list out of the combined list do you think it would be better to have returned separate lists for each action from the start?

That also would lessen the chance in the future of someone not checking the action when creating some new bulk operation on this object list.

Maybe even keep the action bit out of the template engine and split them out into groups in a later step?

What do you think?

Reply:
I was thinking of maintaining separate lists for each "action" when I initially implemented the pruning functionality. However, I discontinued this idea thinking that returning dedicated lists for each “action” would be more cumbersome than having a common “rendered” list with each Object specifying its intended "action". 

Re:
> Maybe even keep the action bit out of the template engine and split them out into groups in a later step?

This will be tricky when dealing with node-level suppressions and pruning: it would require having to fetch the install template `ConfigMaps`, re-render and match against all the items listed for suppression/pruning for each node… that’s why I moved the “RenderAction” to the template engine…

Reply:
Refactored the code to avoid handling the `actionType` in the controller logic...